tests/python/test_image_io.py

```diff
@@ -1,16 +1,38 @@
 import taichi as ti
 import numpy as np
+import pytest
 import os
 
 
+# jpg is also supported but hard to test here since it's lossy:
+@pytest.mark.parametrize('comp,ext', [(3, 'bmp'), (1, 'png'), (3, 'png'),
+                                      (4, 'png')])
+@pytest.mark.parametrize('resx,resy', [(201, 173)])
+@pytest.mark.parametrize('is_tensor', [False, True])
+@pytest.mark.parametrize('dt', [ti.u8])
 @ti.host_arch_only
-def test_image_io():
-    pixel = (np.random.rand(201, 173, 3) * 255).astype(np.uint8)
-    for ext in [
-            'bmp', 'png'
-    ]:  # jpg is also supported but hard to test here since it's lossy
-        fn = 'taichi-image-io-test.' + ext
+def test_image_io(resx, resy, comp, ext, is_tensor, dt):
+    from tempfile import mkstemp
+    if comp != 1:
+        shape = (resx, resy, comp)
+    else:
+        shape = (resx, resy)
+    if is_tensor:
+        pixel_t = ti.var(dt, shape)
```

Comment:
This function is very amazing. it looks so well-designed! :D

LGTM and few nits in my arrogant opinion. Do you think this function has been too **complex** for testing? 

When I wrote ``test_offset.py`` not long ago, our core developers all told me that a test function shoud be as simple as possible. How do you think dividing it into some more parts and test other Taichi tensor such as ``ti.Matrix`` and ``ti.Vector`` as well?

Reply:
> ti.Matrix and ti.Vector

Good idea :) Will add tmr!